### ISSUE TITLE
`return` inserted before `var`

### ISSUE THREAD

**Issue by eventualbuddha** [eventualbuddha/decaffeinate#118]
[This code yields bad JavaScript code:](http://eventualbuddha.github.io/decaffeinate/repl/#?evaluate=false&code=-%3E%0A%20%20a%20%3F%3D%201)

```coffee
->
  a ?= 1
```

```js
(function() {
  if ((typeof a !== "undefined" && a !== null)) { return a; } else { return var a = 1; }
});
```

**Comment by eventualbuddha** [comment]
This CoffeeScript code is actually invalid since you can't `?=` assign a binding that hasn't already been assigned to. That said, this is valid and still triggers it:

```coffee
(a=0) ->
  a ?= 1
```

```js
(function(a=0) {
  if ((typeof a !== "undefined" && a !== null)) { return a; } else { return var a = 1; }
});
```

**Comment by eventualbuddha** [comment]
Fixed in v1.44.12.
